hubstaff-export.rb

```diff
@@ -32,29 +33,275 @@
 require 'date'
 require 'net/http'
 require 'fileutils'
+require 'forwardable'
+require 'base64'
 require 'json'
 require 'pp'
 
-class Hash
-  def compact
-    delete_if {|k,v| v.nil? }
+# a network error occurred
+class NetworkError < StandardError; end
+
+# an error from a command
+class CommandError < StandardError; end
+
+# an error from a command
+class TokenError < StandardError; end
+
+# Simple network wrapper around Net::HTTP
+class HTTPClient
```

Comment:
Why do we need a custom HTTP client? Can we use [faraday](https://lostisland.github.io/faraday/) instead?

Reply:
I'm trying to keep the dependencies to ZERO on this tool (for ease of use for customers)